int64 desired_state
